add stopwatch functionality + custom wait btn

solomiiasabat opened the pull request:
Create a Stopwatch that counts time in HH:MM:SS format.

Stopwatch should have following buttons:
Start/Stop
Wait (works on doubleclick, time between clicks should be 300ms or smaller)
Reset

(!) 300 ms - it’s not a doubleclick event

To pass this task, please use:
Angular Material (design is up to you)
Observables
RxJS

solomiiasabat committed:
add stopwatch functionality + custom wait btn

solomiiasabat committed:
add Angular Material lib, improve app design

solomiiasabat committed:
add unsubscribtion, reduce the number of variables

AleaJactaEst reviewed (commented):
Thanks for following almost everywhere code style guide, but still few things done not according to it :( 
1. Have you checked Angular pipe? I would suggest to use  {{ time | pipeName: format }} in html. 
2. Please try to avoid if (smth) { bla bla bla - you can do it in more functional style with RxJS filter, check line 97 component.ts
3. Probably **critical one!!!** Each time on toggle you will create new subscription - which will be there in memory and running and doing everything in subscription util component destroy - that not a way how it suppose to be. You shouldn't recreate subscription each time..

solomiiasabat committed:
add BehaviorSubject, moved datePipe to template

solomiiasabat commented:
@AleaJactaEst thanks for your feedback! I refactored my code according to your comments:

- moved DatePipe logic from the component to the HTML template;
- used filter operator from RxJS instead of if;
- and used BehaviorSubject from RxJS. It really helped to simplify my code and get rid of unnecessary variables and unnecessary subscriptions.
